### PR TITLE
=doc show the props(() -> new) style in props as the thing to do

### PR DESCRIPTION
**Pull request by ktoso**
More consistently with the scala guide as well (which uses `Props(new Blabla)`)

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/tutorial_1/ActorHierarchyExperiments.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public Receive createReceive() {
 
 //#supervise
 class SupervisingActor extends AbstractActor {
-  ActorRef child = getContext().actorOf(Props.create(SupervisedActor.class), "supervised-actor");
+  ActorRef child = getContext().actorOf(Props.create(() -> new SupervisedActor()), "supervised-actor");
```
Comment (patriknw) [inline akka-docs/src/test/java/jdocs/tutorial_1/ActorHierarchyExperiments.java]: I agree that `new` is better, but then we should use the static `SupervisedActor.props()` factory method convention instead. It's not only about serialization but also that the factory runs in a separate thread and this introduces the risk of accessing something that is not thread-safe from the enclosing scope (parent actor), and that is solved by the static factory method.

**Review by patriknw — CHANGES_REQUESTED** [review]

File: akka-docs/src/test/java/jdocs/tutorial_1/ActorHierarchyExperiments.java (hunk 1 above)
Comment (ktoso) [reply akka-docs/src/test/java/jdocs/tutorial_1/ActorHierarchyExperiments.java]: Okey, will refactor to those after done with persistence

Comment (TimMoore) [reply akka-docs/src/test/java/jdocs/tutorial_1/ActorHierarchyExperiments.java]: Also a better Java convention is to use method references when possible (`SupervisedActor::new` or `SupervisedActor::props`).

**Comment by ktoso** [comment]
Reworking this

**Comment by ktoso** [comment]
Sorry for not finishing this up, would someone want to pick it up or shall I close? @patriknw @jrudolph @johanandren @raboof @chbatey

**Comment by patriknw** [comment]
I can take it from here.

**Comment by ktoso** [comment]
Thanks, and no worries about attribution; Code has moved quite a bit since I think... May be easier to search-replace the same changes globally in a new PR rather than continue this one.

Thanks!

**Comment by patriknw** [comment]
Superseded by https://github.com/akka/akka/pull/25713

**Pull request closed without merging**